tests/failpoints/cases/test_replica_read.rs

```diff
@@ -354,6 +354,44 @@ fn test_read_after_cleanup_range_for_snap() {
     rx1.recv_timeout(Duration::from_secs(5)).unwrap();
 }
 
+#[test]
+fn test_new_split_learner_can_not_find_leader() {
+    let mut cluster = new_node_cluster(0, 4);
+    configure_for_lease_read(&mut cluster, Some(5000), None);
+
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+    let region_id = cluster.run_conf_change();
+
+    cluster.must_put(b"k1", b"v1");
+    cluster.must_put(b"k5", b"v5");
+    pd_client.must_add_peer(region_id, new_peer(2, 2));
+    pd_client.must_add_peer(region_id, new_learner_peer(3, 3));
+    pd_client.must_add_peer(region_id, new_peer(4, 4));
+    for id in 1..=4 {
+        must_get_equal(&cluster.get_engine(id), b"k5", b"v5");
+    }
+
+    fail::cfg("apply_before_split_1_3", "pause").unwrap();
+
+    let region = cluster.get_region(b"k3");
+    cluster.must_split(&region, b"k3");
+    cluster.must_put(b"k2", b"v2");
```

Comment:
How to know it's not the put message that informs learner leadership?

Reply:
The learner is paused at apply split command, so the learner peer of the new split region is not create yet.

Reply:
Can you also put it into comment?

Reply:
Added